.github/workflows/cancel-duplicate-runs.yaml

```diff
@@ -0,0 +1,14 @@
+name: Cancel
+on:
+  workflow_run:
+    workflows: ["CI", "CI Additional", "CI Upstream"]
+    types:
+      - requested
```

Comment:
> I'm not sure I understand when this special workflow is run, though. Before the creation of the individual jobs, maybe?

Yes, this workflow will run when before the creation of jobs defined in `["CI", "CI Additional", "CI Upstream"]` workflows.